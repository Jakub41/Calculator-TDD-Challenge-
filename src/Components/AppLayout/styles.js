import styled from "@emotion/styled";

export const ContentLayout = styled.div`
	width: 100px;
	height: 100px;
`
