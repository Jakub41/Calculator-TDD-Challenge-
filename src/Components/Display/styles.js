import styled from "@emotion/styled";

export const DisplayDiv = styled.div`

`;
