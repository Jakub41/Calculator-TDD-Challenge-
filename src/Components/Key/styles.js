import styled from "@emotion/styled";

export const KeyDiv = styled.div`

`;

export const KeyButton = styled.button`

`;

