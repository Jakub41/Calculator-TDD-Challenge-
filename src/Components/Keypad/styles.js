import styled from "@emotion/styled";

export const KeypadDiv = styled.div`

`;

export const NumbersDiv = styled.div`

`;

export const OperatorsDiv = styled.div`

`;
