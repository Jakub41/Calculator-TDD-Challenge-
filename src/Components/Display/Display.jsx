import React from "react";
import PropTypes from "prop-types";
import { DisplayDiv } from "./styles";

const Display = ({ displayValue }) => (
	<>
		<DisplayDiv>{displayValue}</DisplayDiv>
	</>
);

Display.propTypes = { displayValue: PropTypes.string.isRequired };

export default Display;
