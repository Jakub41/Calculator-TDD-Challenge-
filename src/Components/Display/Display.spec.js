import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Display from "./Display";
import { DisplayDiv } from "./styles";

describe("Display", () => {
	let wrapper;
	const displayComponent = <Display displayValue={""} />;

	beforeEach(() => (wrapper = shallow(displayComponent)));

	it("should render correctly", () => expect(wrapper).toMatchSnapshot());

	it("Display Snapshot renderer", () => {
		const component = renderer.create(displayComponent);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("DisplayDiv Snapshot renderer", () => {
		const styledComponent = renderer.create(<DisplayDiv />);
		let tree = styledComponent.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("should render a <DisplayDiv />", () => {
		expect(wrapper.find(DisplayDiv)).toHaveLength(1);
	});

	it("renders the value of displayValue", () => {
		wrapper.setProps({ displayValue: "test" });
		expect(wrapper.text()).toEqual("test");
	});
});
