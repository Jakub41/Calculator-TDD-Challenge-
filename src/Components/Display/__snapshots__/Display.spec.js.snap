// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Display Snapshot renderer 1`] = `
<div
  className="css-1esthja"
>
  
</div>
`;

exports[`Display DisplayDiv Snapshot renderer 1`] = `
<div
  className="css-1esthja"
/>
`;

exports[`Display should render correctly 1`] = `
<Fragment>
  <Styled(div) />
</Fragment>
`;
